### PR TITLE
Editor: Define the labels of the pattern categories taxonomy

### PR DESCRIPTION
**Pull request by youknowriad**
Trac ticket: https://core.trac.wordpress.org/ticket/60322

In WordPress 6.5, the editor UI is going to be using a standard component for the taxonomies in the editor including the pattern categories. This means that all the labels need to be defined properly for this taxonomy.

This backports the change from https://github.com/WordPress/gutenberg/pull/57094

### CODE REVIEW

**Review by ntsekouras — APPROVED** [review]
✅ Looks good, thank you!

**Comment by youknowriad** [comment]
commit https://core.trac.wordpress.org/changeset/57334

File: src/wp-includes/taxonomy.php [hunk 1]
```diff
@@ -231,8 +231,25 @@ function create_initial_taxonomies() {
...
+				'item_link'                  => __( 'Pattern Category Link' ),
+				'item_link_description'      => __( 'A link to a pattern category.' ),
+				'items_list'                 => __( 'Pattern Categories list' ),
+				'items_list_navigation'      => __( 'Pattern Categories list navigation' ),
+				'new_item_name'              => __( 'New Pattern Category Name' ),
+				'no_terms'                   => __( 'No pattern categories' ),
+				'not_found'                  => __( 'No pattern categories found.' ),
+				'popular_items'              => __( 'Popular Pattern Categories' ),
+				'search_items'               => __( 'Search Pattern Categories' ),
+				'separate_items_with_commas' => __( 'Separate pattern categories with commas' ),
```
Comment (mukeshpanchal27) [inline src/wp-includes/taxonomy.php:250]: For "pattern categories", we use different text cases that need to be consistent.

Comment (swissspidy) [reply src/wp-includes/taxonomy.php:250]: `separate_items_with_commas` is lowercase for regular tags by default, so this seems correct.

Comment (youknowriad) [reply src/wp-includes/taxonomy.php:250]: Yes, I did copy from the tags I think and adapted the labels.

Comment (mukeshpanchal27) [reply src/wp-includes/taxonomy.php:250]: For Link Category taxonomy we use `'back_to_items' => __( '&larr; Go to Link Categories' ),` and for Pattern Category we use `'back_to_items' => __( '&larr; Go to pattern categories' ),` it should be `'back_to_items' => __( '&larr; Go to Pattern Categories' ),`

Comment (youknowriad) [reply src/wp-includes/taxonomy.php:250]: Here's a follow-up https://github.com/WordPress/wordpress-develop/pull/5930

**Pull request closed without merging**